spring-boot-project/spring-boot-actuator/src/main/java/org/springframework/boot/actuate/scheduling/QuartzJobEndpoint.java

```diff
@@ -0,0 +1,190 @@
+/*
+ * Copyright 2012-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot.actuate.scheduling;
+
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import org.quartz.Job;
+import org.quartz.JobDataMap;
+import org.quartz.JobDetail;
+import org.quartz.JobKey;
+import org.quartz.Scheduler;
+import org.quartz.SchedulerException;
+import org.quartz.Trigger;
+import org.quartz.Trigger.TriggerState;
+import org.quartz.TriggerKey;
+import org.quartz.impl.matchers.GroupMatcher;
+
+import org.springframework.boot.actuate.endpoint.annotation.DeleteOperation;
+import org.springframework.boot.actuate.endpoint.annotation.Endpoint;
+import org.springframework.boot.actuate.endpoint.annotation.ReadOperation;
+import org.springframework.boot.actuate.endpoint.annotation.Selector;
+import org.springframework.boot.actuate.endpoint.annotation.WriteOperation;
+import org.springframework.util.Assert;
+
+/**
+ * Web {@link Endpoint} that provides access to the {@link Scheduler} {@link Job}.
+ *
+ * @author Jordan Couret
+ * @since 2.2.0
+ */
+@Endpoint(id = "quartzjob")
+public class QuartzJobEndpoint {
+
+	private final Scheduler scheduler;
+
+	public QuartzJobEndpoint(Scheduler scheduler) {
+		Assert.notNull(scheduler, "Scheduler must not be null");
+		this.scheduler = scheduler;
+	}
+
+	@ReadOperation
+	public Set<JobKey> findAll() throws SchedulerException {
+		return this.scheduler.getJobGroupNames().stream().map((groupName) -> {
+			try {
+				return this.scheduler.getJobKeys(GroupMatcher.groupEquals(groupName));
+			}
+			catch (SchedulerException ignored) {
+				// Do nothing
+				return null;
+			}
+		}).filter(Objects::nonNull).flatMap(Set::stream).collect(Collectors.toSet());
```

Comment:
To get rid of `null`s and `null-check`s
```suggestion
				return Collections.EMPTY_SET;
			}
		}).flatMap(Set::stream).collect(Collectors.toSet());
```